[CI] PolicyManagerTests testFilesEntitlementsWithExclusive failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #6673 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/276747x7q54fc)
- [elasticsearch-pull-request #59261 / part-1](https://gradle-enterprise.elastic.co/s/n4ej676iheaqi)
- [elasticsearch-pull-request #59261 / part-1-entitlements](https://gradle-enterprise.elastic.co/s/4xa526w2wfvgw)

**Reproduction Line:**
```
gradlew ":libs:entitlement:test" --tests "org.elasticsearch.entitlement.runtime.policy.PolicyManagerTests.testFilesEntitlementsWithExclusive" -Dtests.seed=4BF1819C7510CB99 -Dtests.locale=kln -Dtests.timezone=Asia/Hovd -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.entitlement.runtime.policy.PolicyManagerTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testFilesEntitlementsWithExclusive),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [main] 3 failures in test testFilesEntitlementsWithExclusive (2.0% fail rate in 151 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/libs/entitlement/src/test/java/org/elasticsearch/entitlement/runtime/policy/PolicyManagerTests.java
+++ b/libs/entitlement/src/test/java/org/elasticsearch/entitlement/runtime/policy/PolicyManagerTests.java
@@ -429,10 +429,7 @@ public class PolicyManagerTests extends ESTestCase {
                                 "test",
                                 List.of(
                                     new FilesEntitlement(
-                                        List.of(
-                                            FilesEntitlement.FileData.ofPath(testPath1, FilesEntitlement.Mode.READ)
-                                                .withExclusive(true)
-                                        )
+                                        List.of(FilesEntitlement.FileData.ofPath(testPath1, FilesEntitlement.Mode.READ).withExclusive(true))
                                     )
                                 )
                             )
@@ -446,10 +443,7 @@ public class PolicyManagerTests extends ESTestCase {
                                 "test",
                                 List.of(
                                     new FilesEntitlement(
-                                        List.of(
-                                            FilesEntitlement.FileData.ofPath(testPath1, FilesEntitlement.Mode.READ)
-                                                .withExclusive(true)
-                                        )
+                                        List.of(FilesEntitlement.FileData.ofPath(testPath1, FilesEntitlement.Mode.READ).withExclusive(true))
                                     )
                                 )
                             )
@@ -477,8 +471,7 @@ public class PolicyManagerTests extends ESTestCase {
                             List.of(
                                 new FilesEntitlement(
                                     List.of(
-                                        FilesEntitlement.FileData.ofPath(testPath2, FilesEntitlement.Mode.READ)
-                                            .withExclusive(true),
+                                        FilesEntitlement.FileData.ofPath(testPath2, FilesEntitlement.Mode.READ).withExclusive(true),
                                         FilesEntitlement.FileData.ofPath(baseTestPath, FilesEntitlement.Mode.READ)
                                     )
                                 )
@@ -496,10 +489,7 @@ public class PolicyManagerTests extends ESTestCase {
                                 "test",
                                 List.of(
                                     new FilesEntitlement(
-                                        List.of(
-                                            FilesEntitlement.FileData.ofPath(testPath1, FilesEntitlement.Mode.READ)
-                                                .withExclusive(true)
-                                        )
+                                        List.of(FilesEntitlement.FileData.ofPath(testPath1, FilesEntitlement.Mode.READ).withExclusive(true))
                                     )
                                 )
                             )
@@ -514,8 +504,12 @@ public class PolicyManagerTests extends ESTestCase {
             )
         );
         assertEquals(
-            Strings.format("duplicate/overlapping exclusive paths found in files entitlements: "
-                + "[[plugin1] [test] [%s]] and [[(server)] [test] [%s]]", testPath1, testPath2),
+            Strings.format(
+                "duplicate/overlapping exclusive paths found in files entitlements: "
+                    + "[[plugin1] [test] [%s]] and [[(server)] [test] [%s]]",
+                testPath1,
+                testPath2
+            ),
             iae.getMessage()
         );
     }

--- a/libs/entitlement/src/test/java/org/elasticsearch/entitlement/runtime/policy/PolicyManagerTests.java
+++ b/libs/entitlement/src/test/java/org/elasticsearch/entitlement/runtime/policy/PolicyManagerTests.java
@@ -10,6 +10,7 @@
 package org.elasticsearch.entitlement.runtime.policy;
 
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.Strings;
 import org.elasticsearch.entitlement.runtime.policy.PolicyManager.ModuleEntitlements;
 import org.elasticsearch.entitlement.runtime.policy.agent.TestAgent;
 import org.elasticsearch.entitlement.runtime.policy.agent.inner.TestInnerAgent;
@@ -411,6 +412,9 @@ public class PolicyManagerTests extends ESTestCase {
     }
 
     public void testFilesEntitlementsWithExclusive() {
+        var baseTestPath = Path.of("/tmp").toAbsolutePath();
+        var testPath1 = Path.of("/tmp/test").toAbsolutePath();
+        var testPath2 = Path.of("/tmp/test/foo").toAbsolutePath();
         var iae = expectThrows(
             IllegalArgumentException.class,
             () -> new PolicyManager(
@@ -426,7 +430,7 @@ public class PolicyManagerTests extends ESTestCase {
                                 List.of(
                                     new FilesEntitlement(
                                         List.of(
-                                            FilesEntitlement.FileData.ofPath(Path.of("/tmp/test"), FilesEntitlement.Mode.READ)
+                                            FilesEntitlement.FileData.ofPath(testPath1, FilesEntitlement.Mode.READ)
                                                 .withExclusive(true)
                                         )
                                     )
@@ -443,7 +447,7 @@ public class PolicyManagerTests extends ESTestCase {
                                 List.of(
                                     new FilesEntitlement(
                                         List.of(
-                                            FilesEntitlement.FileData.ofPath(Path.of("/tmp/test"), FilesEntitlement.Mode.READ)
+                                            FilesEntitlement.FileData.ofPath(testPath1, FilesEntitlement.Mode.READ)
                                                 .withExclusive(true)
                                         )
                                     )
@@ -460,7 +464,7 @@ public class PolicyManagerTests extends ESTestCase {
             )
         );
         assertTrue(iae.getMessage().contains("duplicate/overlapping exclusive paths found in files entitlements:"));
-        assertTrue(iae.getMessage().contains("[test] [/tmp/test]]"));
+        assertTrue(iae.getMessage().contains(Strings.format("[test] [%s]]", testPath1.toString())));
 
         iae = expectThrows(
             IllegalArgumentException.class,
@@ -473,9 +477,9 @@ public class PolicyManagerTests extends ESTestCase {
                             List.of(
                                 new FilesEntitlement(
                                     List.of(
-                                        FilesEntitlement.FileData.ofPath(Path.of("/tmp/test/foo"), FilesEntitlement.Mode.READ)
+                                        FilesEntitlement.FileData.ofPath(testPath2, FilesEntitlement.Mode.READ)
                                             .withExclusive(true),
-                                        FilesEntitlement.FileData.ofPath(Path.of("/tmp/"), FilesEntitlement.Mode.READ)
+                                        FilesEntitlement.FileData.ofPath(baseTestPath, FilesEntitlement.Mode.READ)
                                     )
                                 )
                             )
@@ -493,7 +497,7 @@ public class PolicyManagerTests extends ESTestCase {
                                 List.of(
                                     new FilesEntitlement(
                                         List.of(
-                                            FilesEntitlement.FileData.ofPath(Path.of("/tmp/test"), FilesEntitlement.Mode.READ)
+                                            FilesEntitlement.FileData.ofPath(testPath1, FilesEntitlement.Mode.READ)
                                                 .withExclusive(true)
                                         )
                                     )
@@ -510,8 +514,8 @@ public class PolicyManagerTests extends ESTestCase {
             )
         );
         assertEquals(
-            "duplicate/overlapping exclusive paths found in files entitlements: "
-                + "[[plugin1] [test] [/tmp/test]] and [[(server)] [test] [/tmp/test/foo]]",
+            Strings.format("duplicate/overlapping exclusive paths found in files entitlements: "
+                + "[[plugin1] [test] [%s]] and [[(server)] [test] [%s]]", testPath1, testPath2),
             iae.getMessage()
         );
     }

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -329,9 +329,6 @@ tests:
 - class: org.elasticsearch.action.admin.cluster.node.tasks.CancellableTasksIT
   method: testChildrenTasksCancelledOnTimeout
   issue: https://github.com/elastic/elasticsearch/issues/123568
-- class: org.elasticsearch.entitlement.runtime.policy.PolicyManagerTests
-  method: testFilesEntitlementsWithExclusive
-  issue: https://github.com/elastic/elasticsearch/issues/123657
 - class: org.elasticsearch.test.rest.yaml.RcsCcsCommonYamlTestSuiteIT
   method: test {p0=search.vectors/41_knn_search_bbq_hnsw/Test knn search}
   issue: https://github.com/elastic/elasticsearch/issues/123727

